grpc/shared/rollupdata_handling.go

```diff
@@ -0,0 +1,266 @@
+	if err != nil {
+		return nil, WrapError(err, fmt.Sprintf("failed to encode public key to bech32m address: %s", publicKey))
+	}
+
+	if auctioneerBech32Address != bech32Address {
+		allocationsWithInvalidPubKey.Inc(1)
+		return nil, fmt.Errorf("address in allocation does not match auctioneer address. expected: %s, got: %s", auctioneerBech32Address, bech32Address)
+	}
+
+	message, err := proto.Marshal(bid)
+	if err != nil {
+		return nil, WrapError(err, "failed to marshal allocation to verify signature")
+	}
+
+	signature := allocation.GetSignature()
+	if !ed25519.Verify(publicKey, message, signature) {
+		allocationsWithInvalidSignature.Inc(1)
+		return nil, fmt.Errorf("signature in allocation is invalid")
+	}
+
+	log.Debug("Allocation is valid. Unmarshalling the transactions in the bid.")
+	// unmarshall the transactions in the bid
+	for _, allocationTx := range bid.GetTransactions() {
+		ethtx := new(types.Transaction)
+		err := ethtx.UnmarshalBinary(allocationTx)
+		if err != nil {
+			return nil, WrapError(err, "failed to unmarshall allocation transaction")
+		}
+		processedTxs = append(processedTxs, ethtx)
+	}
+
+	successfulUnbundledAllocations.Inc(1)
+
+	return processedTxs, nil
+
+}
+
+// `UnbundleRollupDataTransactions` takes in a list of rollup data transactions and returns the corresponding
+// list of Ethereum transactions.
+// If it finds any `Allocation` type, it validates it and places the txs in the `Allocation` at the top of block.
+// Note that `UnbundleRollupDataTransactions` does not return any error on an invalid `RollupData`. If we find any invalid
+// `RollupData` we log the error and continue processing the rest of the transactions. We do not want to break control flow
+// for an invalid transaction as we do not want to interrupt block production.
+// TODO - This function has too many arguments. We should consider refactoring it.
+func UnbundleRollupDataTransactions(txs []*sequencerblockv1.RollupData, height uint64, bridgeAddresses map[string]*params.AstriaBridgeAddressConfig,
+	bridgeAllowedAssets map[string]struct{}, prevBlockHash []byte, auctioneerBech32Address string, auctioneerStartHeight uint64, addressPrefix string) types.Transactions {
+
+	processedTxs := types.Transactions{}
+	allocationTxs := types.Transactions{}
+
+	foundAllocation := false
+
+	for _, tx := range txs {
+		if deposit := tx.GetDeposit(); deposit != nil {
+			depositTx, err := validateAndUnmarshalDepositTx(deposit, height, bridgeAddresses, bridgeAllowedAssets)
+			if err != nil {
+				log.Error("failed to validate and unmarshal deposit tx", "error", err)
+				continue
+			}
+
+			processedTxs = append(processedTxs, depositTx)
+		} else {
+			sequenceData := tx.GetSequencedData()
+
+			if !foundAllocation {
+				if height < auctioneerStartHeight {
+					log.Debug("ignoring allocation tx as it is before the auctioneer start height", "height", height, "auctioneerStartHeight", auctioneerStartHeight)
+					continue
+				}
```

Comment:
It seems like this might be out of place? Are we ignoring all non-deposits transactions occuring below the auctioneer start height if an allocation has not been found?

Reply:
yes you are right, i previously added this check after successfully marshaling an allocation which was right since we were sure that the tx was an allocation and not any other non-deposit tx. We shouldn't not ignore any other non-deposit txs if we are below the `auctioneerStartHeight`. I ll fix this.

Reply:
note that this is not running on testnet and we would have caught this while syncing cause the block hashes with the auctioneer node and the main flame node would be different since we are not executing non-deposit txs

Reply:
addressed here: https://github.com/astriaorg/flame/pull/30/commits/c969f99dd4cd8788573cb927873c766a2d12533a